### PR TITLE
document undocumented fields in Mumble.proto

### LINKED ISSUE #1990: Finish documenting Mumble.proto
The following items have TODOs in `src/Mumble.proto`:

- `Reject.RejectType.None`
- `ServerSync.permissions`
- `UserState.plugin_context`
- `UserState.plugin_identity`
- `BanList.BanEntry.hash`
- `VoiceTarget.Target.group`

### PR DESCRIPTION
**Pull request by bontibon**
fixes #1990

### CODE REVIEW

File: src/Mumble.proto [hunk 1]
```diff
@@ -184,9 +184,14 @@ message UserState {
...
 	optional bytes texture = 11;
-	// TODO ??
+	// The plugin identifier, currently used for positional audio.
+	// Positional audio information is only sent to users who share
+	// identical plugin contexts.
+	//
+	// This value is not trasmitted to clients.
 	optional bytes plugin_context = 12;
-	// TODO ??
+	// The user's plugin-specific identity (currently unused).
```
Comment (hacst) [inline src/Mumble.proto]: This field is used to transmit identity information to the server. See Plugins.cpp:443 . It is never transmitted back to clients though. That much is correct. If you meant that the server isn't processing it itself that's true too but misleading as this information is used and meant to be used by Ice scripts. E.g. https://github.com/mumble-voip/mumo uses it for the bf2 and source functionality. There the identity field contains json with lots of information like team, squad leadership and so on.

File: src/Mumble.proto [hunk 2]
```diff
@@ -63,7 +63,7 @@ message Ping {
 // Sent by the server when it rejects the user connection.
 message Reject {
 	enum RejectType {
-		// TODO ??
+		// The rejection reason is unknown (murmur will never send this value).
```
Comment (hacst) [inline src/Mumble.proto]: If the stock client interprets this as "Unknown" then we shouldn't say the server never sends it. We will likely need it for backwards compatibility in the future and if someone implements theprotocol they should handle it correctly.

File: src/Mumble.proto [hunk 3]
```diff
@@ -184,9 +184,14 @@ message UserState {
 	optional bool self_deaf = 10;
 	// User image if it is less than 128 bytes.
 	optional bytes texture = 11;
-	// TODO ??
+	// The plugin identifier, currently used for positional audio.
```
Comment (hacst) [inline src/Mumble.proto]: Not sure if we would ever could/would use it for anything else ;)

File: src/Mumble.proto [hunk 4]
```diff
@@ -213,7 +218,7 @@ message BanList {
 		required uint32 mask = 2;
 		// User name for identification purposes (does not affect the ban).
 		optional string name = 3;
-		// TODO ??
+		// The certificate of the banned user.
```
Comment (hacst) [inline src/Mumble.proto]: SHA-1 certificate **hash**.

File: src/Mumble.proto [hunk 5]
```diff
@@ -409,7 +414,7 @@ message VoiceTarget {
 		repeated uint32 session = 1;
 		// Channels that are included as targets.
 		optional uint32 channel_id = 2;
-		// TODO ??
+		// ACL groups that are included as targets.
```
Comment (hacst) [inline src/Mumble.proto]: I don't think this field takes multiple groups.

File: src/Mumble.proto [hunk 6]
```diff
@@ -409,7 +414,7 @@ message VoiceTarget {
 		repeated uint32 session = 1;
 		// Channels that are included as targets.
```
Comment (hacst) [inline src/Mumble.proto]: Don't get the plural here.

Comment (hacst) [inline src/Mumble.proto]: Don't get the plural here.

**Comment by hacst** [comment]
Hi. Thanks for the PR. See comments.

**Comment by bontibon** [comment]
@hacst Updated

**Comment by hacst** [comment]
Merged. Thanks.

(that new web-squash and merge github feature is kinda neat)

**Pull request merged**